timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -449,25 +367,73 @@ void __global__ k_nonbonded_unified(
     unsigned long long *__restrict__ du_dx,
     unsigned long long *__restrict__ du_dp,
     unsigned long long *__restrict__ u_buffer) {
-
-    int tile_idx = blockIdx.x;
-    int row_block_idx = ixn_tiles[tile_idx];
-    int index = row_block_idx * 32 + threadIdx.x;
-    const unsigned int atom_i_idx = index < NR ? row_idxs[index] : N;
-    int atom_j_idx = ixn_atoms[tile_idx * 32 + threadIdx.x];
-
-    RealType w_i = atom_i_idx < N ? params[atom_i_idx * PARAMS_PER_ATOM + PARAM_OFFSET_W] : 0;
-    RealType w_j = atom_j_idx < N ? params[atom_j_idx * PARAMS_PER_ATOM + PARAM_OFFSET_W] : 0;
-
-    int is_vanilla = w_i == 0 && w_j == 0;
-
-    bool tile_is_vanilla = __all_sync(0xffffffff, is_vanilla);
-
-    if (tile_is_vanilla) {
-        v_nonbonded_unified<RealType, 0, COMPUTE_U, COMPUTE_DU_DX, COMPUTE_DU_DP>(
-            N, NR, coords, params, box, beta, cutoff, row_idxs, ixn_tiles, ixn_atoms, du_dx, du_dp, u_buffer);
-    } else {
-        v_nonbonded_unified<RealType, 1, COMPUTE_U, COMPUTE_DU_DX, COMPUTE_DU_DP>(
-            N, NR, coords, params, box, beta, cutoff, row_idxs, ixn_tiles, ixn_atoms, du_dx, du_dp, u_buffer);
-    };
+    __shared__ box_cache<RealType> shared_box;
+    if (threadIdx.x == 0) {
```

Comment:
any benefit to parallelizing the stores to 3/6 threads?

Reply:
No difference that I can tell and with the struct becomes messier